gluten-data/src/main/scala/io/glutenproject/utils/ArrowUtil.scala

```diff
@@ -86,4 +87,23 @@ object ArrowUtil extends Logging {
     }
     new Schema(fields)
   }
+
+  /** Convert `ArrowBuf` to `Array[Byte]`, the input `ArrowBuf` is not closed */
+  def toByteArray(buffer: ArrowBuf): Array[Byte] = {
+    val readableBytes = buffer.readableBytes()
+    if (readableBytes > Integer.MAX_VALUE) {
+      throw new GlutenException(
+        s"arrow buf readableBytes: $readableBytes is bigger than Integer.MAX_VALUE")
+    }
+    val buf = buffer.nioBuffer
+    val bytes: Array[Byte] = new Array[Byte](buf.remaining())
+    buf.get(bytes)
+    bytes
```

Comment:
OK so it does conduct copy. This was intentional, right?

Copying also making sense to me as long as it's a intended design in the early version of the feature.

Reply:
yes, I think it's necessary as the Arrow always use direct(offheap) buffer but we need a onheap buffer.